File: CS/WindowsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Grid;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            new DevExpress.XtraGrid.Design.XViewsPrinting(gridControl1);
            gridView1.OptionsSelection.MultiSelect = true;
            gridView1.OptionsBehavior.Editable = false;
            gridView1.OptionsSelection.EnableAppearanceFocusedCell = false;
        }

        private void SelectRows(GridView view, int startRow, int endRow)
        {
            if (startRow > -1 && endRow > -1)
            {
                view.BeginSelection();
                view.ClearSelection();
                view.SelectRange(startRow, endRow);
                view.EndSelection();
            }
        }

        private int GetRowAt(GridView view, int x, int y)
        {
            return view.CalcHitInfo(new Point(x, y)).RowHandle;
        }

        private int StartRowHandle = -1;
        private int CurrentRowHandle = -1;

        private void gridView1_MouseDown(object sender, System.Windows.Forms.MouseEventArgs e)
        {
            StartRowHandle = GetRowAt(sender as GridView, e.X, e.Y);
        }

        private void gridView1_MouseMove(object sender, System.Windows.Forms.MouseEventArgs e)
        {
            int newRowHandle = GetRowAt(sender as GridView, e.X, e.Y);
            if (CurrentRowHandle != newRowHandle)
            {
                CurrentRowHandle = newRowHandle;
                SelectRows(sender as GridView, StartRowHandle, CurrentRowHandle);
            }
        }

        private void gridView1_MouseUp(object sender, System.Windows.Forms.MouseEventArgs e)
        {
            StartRowHandle = -1;
            CurrentRowHandle = -1;
        }
    }
}

File: VB/WindowsApplication1/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Grid

Namespace WindowsApplication1

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            Dim tmp_XViewsPrinting = New DevExpress.XtraGrid.Design.XViewsPrinting(gridControl1)
            gridView1.OptionsSelection.MultiSelect = True
            gridView1.OptionsBehavior.Editable = False
            gridView1.OptionsSelection.EnableAppearanceFocusedCell = False
        End Sub

        Private Sub SelectRows(ByVal view As GridView, ByVal startRow As Integer, ByVal endRow As Integer)
            If startRow > -1 AndAlso endRow > -1 Then
                view.BeginSelection()
                view.ClearSelection()
                view.SelectRange(startRow, endRow)
                view.EndSelection()
            End If
        End Sub

        Private Function GetRowAt(ByVal view As GridView, ByVal x As Integer, ByVal y As Integer) As Integer
            Return view.CalcHitInfo(New Point(x, y)).RowHandle
        End Function

        Private StartRowHandle As Integer = -1

        Private CurrentRowHandle As Integer = -1

        Private Sub gridView1_MouseDown(ByVal sender As Object, ByVal e As MouseEventArgs)
            StartRowHandle = GetRowAt(TryCast(sender, GridView), e.X, e.Y)
        End Sub

        Private Sub gridView1_MouseMove(ByVal sender As Object, ByVal e As MouseEventArgs)
            Dim newRowHandle As Integer = GetRowAt(TryCast(sender, GridView), e.X, e.Y)
            If CurrentRowHandle <> newRowHandle Then
                CurrentRowHandle = newRowHandle
                SelectRows(TryCast(sender, GridView), StartRowHandle, CurrentRowHandle)
            End If
        End Sub

        Private Sub gridView1_MouseUp(ByVal sender As Object, ByVal e As MouseEventArgs)
            StartRowHandle = -1
            CurrentRowHandle = -1
        End Sub
    End Class
End Namespace
